### PR TITLE
container add support for all GA fields in `auto_provisioning_defaults`

### PR DESCRIPTION
**Pull request by dysosmus**
This MR is an upstream from https://github.com/hashicorp/terraform-provider-google/pull/10610. 
It implements (or move to) multiple GA field parts of the `cluster_autoscaling` from the `google_container_cluster` resource.

Do note that some behaviours still need to be better defined:
- `boot_disk_kms_key`, is optional but apparently once set it seems not possible to go back to GCP managed encryption, it however can be changed to another key. 
- `disk_type`, `image_type`, `disk_size_gb`, are all optional and computed but once set there is no way to go back to the initial default. 

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `disk_size_gb`, `disk_type`, `boot_disk_kms_key`, `image_type`,  `shielded_instance_config`, `management`, `upgrade_settings` fields under `cluster_autoscaling.auto_provisioning_defaults` for `google_container_cluster`.
container: moved `cluster_autoscaling.auto_provisioning_defaults.min_cpu_platform` to GA for `google_container_cluster`
container: moved `cluster_autoscaling.autoscaling_profile` to GA for `google_container_cluster`
```

### CODE REVIEW

**Comment by alexbacchin-asx** [comment]
Is there an update on this PR?

**Comment by yannpichon-su** [comment]
Hello @slevenick, we are highly interested in this feature (especially the shielded_instance_config block). Do you have any idea when this will be integrated in the Terraform provider ? Thanks 😉

**Comment by slevenick** [comment]
/gcbrun

Looks like the build is blocked on some Terraform conversion fields. @melinath can you have a look and see what would need to change to get that step to run?

**Comment by slevenick** [comment]
Just kidding, it's a failure in the resource that shows up everywhere.

```
google/resource_container_cluster.go:371:10: duplicate key "image_type" in map literal
	previous key at google/resource_container_cluster.go:337:10
```

Looks like we have a duplicate declaration

**Comment by melinath** [comment]
Unsubscribing - ping me again if needed!

**Comment by wmaramos** [comment]
hey @dysosmus thanks for the contribution, do you believe you can move forward with this PR?

**Comment by slevenick** [comment]
Looks like it still has the same error message:
```
resource_container_cluster.go:371:10: duplicate key "image_type" in map literal
```

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -457,33 +457,125 @@ func resourceContainerCluster() *schema.Resource {
...
+										ValidateFunc: validation.StringInSlice([]string{"pd-standard", "pd-balanced", "pd-ssd"}, false),
+										Description:  `Type of the disk attached to each node. If unspecified, the default disk type is 'pd-standard'`,
+									},
+									"boot_disk_kms_key": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Computed:    true,
+										Description: `The Customer Managed Encryption Key used to encrypt the boot disk attached to each node in the node pool. This should be of the form projects/[KEY_PROJECT_ID]/locations/[LOCATION]/keyRings/[RING_NAME]/cryptoKeys/[KEY_NAME]. For more information about protecting resources with Cloud KMS Keys please see: https://cloud.google.com/compute/docs/disks/customer-managed-encryption.`,
+									},
+									"image_type": {
```
Comment (slevenick) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:495]: This looks to be the offending line

Comment (alexbacchin-asx) [reply mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:495]: @dysosmus can you please check?

Comment (dysosmus) [reply mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:495]: Hey I will check, sorry for the slow reaction time :).

**Pull request closed without merging**